src/ssl.c

```diff
@@ -21307,19 +21307,9 @@ WOLFSSL_X509_STORE* wolfSSL_X509_STORE_new(void)
     return store;
 
 err_exit:
-    if(store == NULL)
+    if (store == NULL)
         return NULL;
-    if(store->cm != NULL)
-        wolfSSL_CertManagerFree(store->cm);
```

Comment:
Removing the Free does not look right at all.  When only this function is called, and failure happens after the CertManager is allocated you've now lost it.  Please create a thorough solution for this function and other functions that use this type.

Reply:
The frees are being performed in wolfSSL_X509_STORE_free() by the call on line 21323/21313 (old/new).
This was my interpretation of the compile warning that reported that there was a potential double free as the deleted lines never set the fields to NULL.

Reply:
I see, thanks.